### PR TITLE
protect against null-path

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR...

Fixes: #8747

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [x] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?

This is not re-produced bug. The fix is fairly simple. I have no idea how to reproduce but it resulted in data loss in a production VM. I think the best we can do is test upgrades from different versions. and see if anything happens.

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Review by vishesh92 — APPROVED** [review]
clgtm

File: plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java [hunk 1]
```diff
@@ -2969,7 +2969,10 @@ protected void setVAppPropertiesToConfigSpec(VmConfigInfo vAppConfig,
     }
 
     private String appendFileType(String path, String fileType) {
-        if (path.toLowerCase().endsWith(fileType.toLowerCase())) {
+        if (path == null) {
+            throw new CloudRuntimeException("No path given, cannot append filetype" + fileType);
+        }
+        if (fileType = null || path.toLowerCase().endsWith(fileType.toLowerCase())) {
```
Comment (vishesh92) [inline plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java:2975]:
```suggestion
        if (fileType == null || path.toLowerCase().endsWith(fileType.toLowerCase())) {
```

**Review by vishesh92 — CHANGES_REQUESTED** [review]
@DaanHoogland a small typo.

File: plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java (hunk 1 above)
Comment (DaanHoogland) [reply plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java:2975]: :doh:

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan test

**Review by Pearl1594 — APPROVED** [review]
code LGTM

File: plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java [hunk 2]
```diff
@@ -2969,7 +2969,10 @@ protected void setVAppPropertiesToConfigSpec(VmConfigInfo vAppConfig,
     }
 
     private String appendFileType(String path, String fileType) {
-        if (path.toLowerCase().endsWith(fileType.toLowerCase())) {
+        if (path == null) {
```
Comment (sureshanaparti) [inline plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java:2972]:
```suggestion
        if (StringUtils.isBlank(path)) {
```

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test alma9 vmware-70u3

**Comment by DaanHoogland** [comment]
@blueorangutan test rocky8 vmware-70u3

**Review by vishesh92 — APPROVED** [review]
clgtm

File: plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java [hunk 3]
```diff
@@ -2969,7 +2969,10 @@ protected void setVAppPropertiesToConfigSpec(VmConfigInfo vAppConfig,
     }
 
     private String appendFileType(String path, String fileType) {
-        if (path.toLowerCase().endsWith(fileType.toLowerCase())) {
+        if (StringUtils.isBlank(path)) {
+            throw new CloudRuntimeException("No path given, cannot append filetype" + fileType);
```
Comment (sureshanaparti) [inline plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java:2973]:
```suggestion
            throw new CloudRuntimeException("No path given, cannot append filetype " + fileType);
```

**Review by sureshanaparti — APPROVED** [review]
clgtm

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test alma9 vmware-70u3

**Comment by DaanHoogland** [comment]
@rajujith any findings?

**Review by rajujith — APPROVED** [review]
LGTM. 

I am unable to reproduce the issue resulting in the 'NULL' path for volume, however, based on the logic in the fix the issue should be fixed.

**Pull request merged**